hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/InternalOperationService.java

```diff
@@ -68,7 +67,7 @@
      */
     List<SlowOperationDTO> getSlowOperationDTOs();
 
-    <E> void asyncInvokeOnPartition(String serviceName, Operation op, int partitionId, Callback<E> callback);
+    void asyncInvokeOnPartition(String serviceName, Operation op, int partitionId, Object callback);
```

Comment:
Since these `asyncInvoke*` methods are new additions, why don't you just convert these to use `ExecutionCallback`?
```java
void asyncInvokeOnPartition(String serviceName, Operation op, int partitionId, ExecutionCallback<E> callback)
```

I think there's no need to support `Callback` version...